### ISSUE TITLE
Processing Enter (for example) on a form

### ISSUE THREAD

**Issue by thunderkid** [glortho/react-keydown#75]
I have a form (though I'm not using html native `<form>`), and I'm trying to process the Enter key. I'm decorating my method with `@keydown('enter')` but this only works if focus isn't in an input field. (Though it works fine if I use `'ctrl+enter'` instead).

Is there a nice way to set `forceConsider` so that `_shouldConsider `will let it through even when focus is on an input field? Seems like I need a `@keydownForAnyFocus` decorator or something similar...

Great library! Thanks.

**Comment by glortho** [comment]
@thunderkid yeah we're thinking about including an option to trigger in inputs when you set up the binding. in the meantime, you can do something like:

```jsx
<input onKeyDown={ myDecoratedMethod } />
```
Or if you're using it on a lot of them:
```jsx
const Input = props => <input onKeyDown={ myDecoratedMethod } { ...props } />

<Input type="text">
```

**Comment by glortho** [comment]
let me know if this doesn't work for you and we'll re-open

**Comment by thunderkid** [comment]
Yes, I noticed that approach in your docs. But I didn't like it since in many cases I've got nested components that set up the fields, so sprinkling `onKeyDown`'s around and having to pass the method down through subcomponent layers would get pretty ugly, just to catch an enter key.

Any idea on the timing for releasing this option? I'd offer to take a look and submit a PR, but it seems like it requires some high-level decisions that I'd only be guessing at, so there'd be a 99% chance you'd want to do it differently.

**Comment by glortho** [comment]
Yeah I hear you. And you can't just wrap your inputs in a `<form onSubmit={ myMethod }>` to get the default submit-on-enter behavior?

If not, a PR for this functionality likely wouldn't be too complex. If we started with just method decorators, you could add a condition here that checks for the option and makes sure we have a keydown event: https://github.com/glortho/react-keydown/blob/master/src/decorators/method_decorator.js#L59

That's my best guess anyway. I could probably get to it sometime in the next couple of weeks if you don't want to take it on @thunderkid

**Comment by thunderkid** [comment]
I tried `onSubmit`, but it wasn't great. (Because ... Interfered with other third party react components weirdly across browsers ... even though it handles 'enter', I wouldn't have resolved the problem for any other key I wanted to trap, like 'esc' ... suddenly I'd have two parallel logic routes - one for enter and one for processing control keys.)

Thanks for the suggestions. I'll start poking around with the code, though it's pretty alien to me at this point...

**Comment by thunderkid** [comment]
Please excuse the naive question, but I'm trying to get going on a PR, and I'm not sure whether I'm missing something in your build environment.
I've cloned the repository, and run npm install. Then I run the local webpack with `.\node_modules\.bin\webpack`, which creates only a 2k `example\public\js\app.js` file. When I open `example\public\index.html` in a browser it gives errors. I see no sign of an output js file being create for the actual library, and `webpack.config.js` only seems to reference the examples folders.

So how do I run the build? Is there another webpack config file hidden away somewhere?

**Comment by glortho** [comment]
Hm ok well running `webpack` should build the example for you. What is the
error you get when you open index.html?

**Comment by thunderkid** [comment]
I get `Uncaught syntax error: Unexpected token Import`, and the `example/public/js/app.js` file contains 
```
/***/ (function(module, exports) {

	import React from 'react';
	import ReactDOM from 'react-dom';

	import App from './app';

	ReactDOM.render(React.createElement(App, null), document.getElementById('example'));

/***/ })
/******/ ]);
```

I just tried on a fresh machine with the same result, whether I run my global webpack or the local one under `node_modules\.bin\`.

**Comment by glortho** [comment]
Ah what browser are you using? Try latest Chrome or change `{ "modules":
false }` to `{ "modules": "umd" }` in .babelrc

**Comment by thunderkid** [comment]
I was already using latest Chrome 62.0.3202.75 under windows 10.
But `modules: "umd"` fixed it. Thanks.

**Comment by thunderkid** [comment]
What command do you run to create a library bundle as opposed to the entire example app?

**Comment by glortho** [comment]
Check out our package.json for our build scripts. We do mainly `npm run
build` and `npm run build:es`

**Comment by thunderkid** [comment]
`npm run build:es` works. But `npm run buld` gives `BABEL_ENV is not recognized as an internal or external command`.

**Comment by glortho** [comment]
Ah right you're on Windows. I'm not sure what the equivalent is, but it
should be okay if you can set an environment variable to that value
BABEL_ENV=production. Sorry I'm not too familiar with Windows dev
environments!

**Comment by thunderkid** [comment]
Turns out there's an npm utility `cross-env` that makes `npm build` work in windows. I'm noticing that `npm build` and `npm build:es` are producing binary identical files, ie `BABEL_ENV=production` changes nothing. Does this seem right or is windows still ignoring something?

**Comment by thunderkid** [comment]
So I can make changes and test them in your example app, but now I'm trying to test them in my own app, and I'm having no success running a local copy of the library.

Here's the problem: In my own app I first run the published version: I `npm install react-keydown@1.9.5` and then I use `import keydown, { Keys } from 'react-keydown'`. Everything is fine. I can `console.log(Keys)`, and `@keydown` behaves as expected.

Now for the local version: I `git clone` the exact same version for consistency, run `npm run prepublish` in my local folder, and change my import statement to `import keydown, { Keys } from '/my/local/path/react-keydown'`. Now I can still `console.log(Keys)` ok, but `@keydown` does absolutely nothing, and shows no errors.

I've tried dozens of combinations, eg flipping `"modules": "umd"` back to false, putting in the entire path to the `'dist/index.js'` file, etc, but to no avail. When I build a toy npm library I can access it locally in this way with no problem. And even for `react-keydown` it seems I can see some of it - ie `Keys`. But `@keydown` just refuses to do anything. 

Any ideas? Thanks.
